presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestApproximateCountDistinctBoolean.java

```diff
@@ -0,0 +1,117 @@
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static com.facebook.presto.metadata.FunctionKind.AGGREGATE;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static org.testng.Assert.assertEquals;
+
+public class TestApproximateCountDistinctBoolean
+        extends AbstractTestApproximateCountDistinct
+{
+    @Override
+    public InternalAggregationFunction getAggregationFunction()
+    {
+        return metadata.getFunctionRegistry().getAggregateFunctionImplementation(
+                new Signature("approx_distinct", AGGREGATE, BIGINT.getTypeSignature(), BOOLEAN.getTypeSignature(), DOUBLE.getTypeSignature()));
+    }
+
+    @Override
+    public Type getValueType()
+    {
+        return BOOLEAN;
+    }
+
+    @Override
+    public Object randomValue()
+    {
+        return (boolean) ThreadLocalRandom.current().nextBoolean();
+    }
+
+    @DataProvider(name = "inputSequences")
+    public Object[][] inputSequences()
+    {
+        return new Object[][] {
+                {true},
+                {false},
+                {true, true, true},
+                {false, false, false},
+                {true, false, true, false},
+        };
+    }
+
+    @Test(dataProvider = "inputSequences")
+    public void testMixed(boolean... inputSequence)
+    {
+        List<Object> booleanList = new ArrayList<>();
+        for (boolean b : inputSequence) {
+            booleanList.add(b);
+        }
+        assertEquals(distinctCount(booleanList), estimateCount(booleanList, 1));
+    }
+
+    private int distinctCount(List<Object> inputSequence)
+    {
+        boolean sawZero = false;
+        boolean sawOne = false;
+        if (inputSequence == null || inputSequence.isEmpty()) {
+            return 0;
+        }
+        for (int i = 0; i < inputSequence.size(); i++) {
+            sawOne = (Boolean) inputSequence.get(i) ? true : sawOne;
+            sawZero = (Boolean) inputSequence.get(i) ? sawZero : true;
+        }
+        if (!(sawZero || sawOne)) {
+            return 0;
+        }
+        else if (sawZero && sawOne) {
+            return 2;
+        }
+        else {
+            return 1;
+        }
+    }
+
+    protected Page createPage(List<Object> values, double maxStandardError)
+    {
+        if (values.isEmpty()) {
+            return new Page(0);
+        }
+        else {
+            return new Page(values.size(),
+                    createBlock(getValueType(), values),
+                    createBlock(BOOLEAN, ImmutableList.copyOf(Collections.nCopies(values.size(), maxStandardError))));
```

Comment:
I don't think the `BOOLEAN` type here is correct. This block contains the required standard error which is always a `DOUBLE`. Also, I don't think this function is needed (see my comment above).